simpl/dicts.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) 2011-2015 Rackspace US, Inc.
+# All Rights Reserved.
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+"""Helpers for built-in `dict` class."""
+
+
+def write_path(target, path, value, separator='/'):
+    """Write a value deep into a dict building any intermediate keys.
+
+    :param target: a dict to write data to
+    :param path: a key or path to a key (path is delimited by `separator`)
+    :param value: the value to write to the key
+    :keyword separator: the separator used in the path (ex. Could be "." for a
+        json/mongodb type of value)
+    """
+    parts = path.split(separator)
+    current = target
+    for part in parts[:-1]:
+        if part not in current:
+            current[part] = current = {}
+        else:
+            current = current[part]
+    current[parts[-1]] = value
+
+
+def read_path(source, path, separator='/'):
+    """Read a value from a dict supporting a deep path as a key.
+
+    :param source: a dict to read data from
+    :param path: a key or path to a key (path is delimited by `separator`)
+    :keyword separator: the separator used in the path (ex. Could be "." for a
+        json/mongodb type of value)
+    """
+    parts = path.strip(separator).split(separator)
+    current = source
+    for part in parts[:-1]:
+        if part not in current:
+            return
```

Comment:
Please add a test which exercises this `return` case. It seems a bit like an error/edge condition.

This behavior should also be document in the docstring: if a path does not exist, return None (as opposed to a `KeyError`).